CA2235 int is not serializable

I have the latest public releases of the analyser packages:

```xml
    <GlobalPackageReference Include="StyleCop.Analyzers" Version="1.0.2" IncludeAssets="Analyzers;Build" PrivateAssets="All" />
    <GlobalPackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="2.6.3" IncludeAssets="Analyzers;Build" PrivateAssets="All" />
    <GlobalPackageReference Include="Microsoft.CodeAnalysis" Version="2.10.0" IncludeAssets="Analyzers;Build" PrivateAssets="All" />
    <GlobalPackageReference Include="Microsoft.Net.Compilers" Version="2.10.0" IncludeAssets="Analyzers;Build" PrivateAssets="All" />
```

and I'm getting the error:

```
 error CA2235: Field LineNumber is a member of type LineInfo which is serializable but is of type int which is not serializable 
```

all related issues I've found have been closed, so I assume it's expected to be fixed in the most recent versions?

--- a/src/Microsoft.NetCore.Analyzers/Core/Runtime/SerializationRulesDiagnosticAnalyzer.cs
+++ b/src/Microsoft.NetCore.Analyzers/Core/Runtime/SerializationRulesDiagnosticAnalyzer.cs
@@ -266,6 +266,11 @@ namespace Microsoft.NetCore.Analyzers.Runtime
 
             private bool IsSerializable(ITypeSymbol type)
             {
+                if (type.IsPrimitiveType())
+                {
+                    return true;
+                }
+
                 switch (type.TypeKind)
                 {
                     case TypeKind.Array:

